--- a/node_modules/ansi-styles/index.js
+++ b/node_modules/ansi-styles/index.js
@@ -162,4 +162 @@ function assembleStyles() {
-Object.defineProperty(module, 'exports', {
-	enumerable: true,
-	get: assembleStyles
-});
+module.exports = assembleStyles();
--- a/node_modules/form-data/lib/form_data.js
+++ b/node_modules/form-data/lib/form_data.js
@@ -9 +9 @@ var https = require('https');
-var asynckit = require('asynckit');
+var asynckit = {parallel: require('asynckit/parallel')};
--- a/node_modules/is-windows/index.js
+++ b/node_modules/is-windows/index.js
@@ -8 +8 @@
-(function(factory) {
+/*(function(factory) {
@@ -23,2 +23,2 @@
-  'use strict';
-  return function isWindows() {
+*/'use strict';
+  module.exports = function isWindows() {
@@ -27 +27,0 @@
-});
--- a/node_modules/jsbn/index.js
+++ b/node_modules/jsbn/index.js
@@ -1 +1,0 @@
-(function(){
@@ -1350 +1349 @@
-  if (typeof exports !== 'undefined') {
+  if (true) {
@@ -1357 +1356,0 @@
-}).call(this);
--- a/node_modules/lodash._baseclone/index.js
+++ b/node_modules/lodash._baseclone/index.js
@@ -86 +86 @@
-/** Detect free variable `exports`. */
+/** Detect free variable `exports`.
@@ -91 +91,0 @@
-/** Detect free variable `module`. */
@@ -96 +95,0 @@
-/** Detect the popular CommonJS extension `module.exports`. */
@@ -102 +100 @@ /** Detect free variable `global` from Node.js. */
-var freeGlobal = checkGlobal(freeExports && freeModule && typeof global == 'object' && global);
+var freeGlobal = checkGlobal(true && true && typeof global == 'object' && global);
@@ -322 +320 @@ /** Built-in value references. */
-var Buffer = moduleExports ? root.Buffer : undefined,
+var Buffer = true ? root.Buffer : undefined,
--- a/node_modules/lodash/_cloneBuffer.js
+++ b/node_modules/lodash/_cloneBuffer.js
@@ -3 +3 @@
-/** Detect free variable `exports`. */
+/** Detect free variable `exports`.
@@ -6 +6,0 @@
-/** Detect free variable `module`. */
@@ -9 +8,0 @@
-/** Detect the popular CommonJS extension `module.exports`. */
@@ -13 +11 @@ /** Built-in value references. */
-var Buffer = moduleExports ? root.Buffer : undefined,
+var Buffer = true ? root.Buffer : undefined,
--- a/node_modules/lodash/_nodeUtil.js
+++ b/node_modules/lodash/_nodeUtil.js
@@ -3 +3 @@
-/** Detect free variable `exports`. */
+/** Detect free variable `exports`.
@@ -6 +6,0 @@
-/** Detect free variable `module`. */
@@ -9 +8,0 @@
-/** Detect the popular CommonJS extension `module.exports`. */
@@ -13 +11 @@ /** Detect free variable `process` from Node.js. */
-var freeProcess = moduleExports && freeGlobal.process;
+var freeProcess = true && freeGlobal.process;
@@ -19 +17 @@   try {
-    var types = freeModule && freeModule.require && freeModule.require('util').types;
+    var types = /*freeModule && freeModule.require &&*/ module.require('util').types;
--- a/node_modules/lodash/isBuffer.js
+++ b/node_modules/lodash/isBuffer.js
@@ -4 +4 @@
-/** Detect free variable `exports`. */
+/** Detect free variable `exports`.
@@ -7 +7,0 @@
-/** Detect free variable `module`. */
@@ -10 +9,0 @@
-/** Detect the popular CommonJS extension `module.exports`. */
@@ -14 +12 @@ /** Built-in value references. */
-var Buffer = moduleExports ? root.Buffer : undefined;
+var Buffer = true ? root.Buffer : undefined;
--- a/node_modules/object-path/index.js
+++ b/node_modules/object-path/index.js
@@ -1 +1 @@
-(function (root, factory){
+/*(function (root, factory){
@@ -4 +4,0 @@
-  /*istanbul ignore next:cant test*/
@@ -14 +13 @@ (function (root, factory){
-})(this, function(){
+})(this, function(){*/
@@ -291,2 +276 @@   mod.withInheritedProps = factory({includeInheritedProps: true})
-  return mod;
-});
+  module.exports = mod;
--- a/node_modules/puka/index.js
+++ b/node_modules/puka/index.js
@@ -600,3 +600 @@   static sh(templateSpans) {
-    for (var _len = arguments.length, values = Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
-      values[_key - 1] = arguments[_key];
-    }
+    var values = Array.prototype.slice.call(arguments, 1);
--- a/node_modules/tweetnacl/nacl-fast.js
+++ b/node_modules/tweetnacl/nacl-fast.js
@@ -1 +1 @@
-(function(nacl) {
+/*(function(nacl) {*/ var nacl = module.exports;
@@ -2388 +2388,0 @@
-})(typeof module !== 'undefined' && module.exports ? module.exports : (self.nacl = self.nacl || {}));
--- a/node_modules/yarn/src/cli/commands/config.js
+++ b/node_modules/yarn/src/cli/commands/config.js
@@ -41 +41,3 @@ const CONFIG_KEYS = [
+  'packageDateLimit',
+  'disableWrappersFolder',
 ];
--- a/node_modules/yarn/src/cli/index.js
+++ b/node_modules/yarn/src/cli/index.js
@@ -29 +29 @@
-process.stdout.prependListener('error', err => {
+(process.stdout.prependListener || process.stdout.on)('error', err => {
@@ -137 +137,2 @@ export async function main({
   commander.option('--otp <otpcode>', 'one-time password for two factor authentication');
+  commander.option('--package-date-limit <time>', 'only install package version that have release date before this');
@@ -560 +561,2 @@ export async function main({
       otp: commander.otp,
+      packageDateLimit: commander.packageDateLimit,
--- a/node_modules/yarn/src/config.js
+++ b/node_modules/yarn/src/config.js
@@ -72 +72,3 @@ export type ConfigOptions = {
   otp?: string,
+  packageDateLimit?: ?string,
+  disableWrappersFolder?: boolean,
@@ -207 +209,3 @@ export default class Config {
   otp: ?string;
+  packageDateLimit: ?string;
+  disableWrappersFolder: boolean;
@@ -347 +351,4 @@ export default class Config {
+    this.packageDateLimit = opts.packageDateLimit || String(this.getOption('package-date-limit') || '') || null;
+    this.disableWrappersFolder = Boolean(this.getOption('disable-wrappers-folder'));
+
     this.globalFolder = opts.globalFolder || String(this.getOption('global-folder', true));
@@ -828 +835 @@ export default class Config {
-    for (const file of new Set([].concat(...files))) {
+    for (const file of new Set(Array.prototype.concat.apply([], files))) {
--- a/node_modules/yarn/src/errors.js
+++ b/node_modules/yarn/src/errors.js
@@ -24,2 +24,6 @@ export class ProcessTermError extends MessageError {
-  EXIT_CODE: ?number;
-  EXIT_SIGNAL: ?string;
+  constructor(msg: string, code?: string) {
+    super(msg, code);
+  }
+
+  EXIT_CODE: ?number = undefined;
+  EXIT_SIGNAL: ?string = undefined;
--- a/node_modules/yarn/src/fetchers/tarball-fetcher.js
+++ b/node_modules/yarn/src/fetchers/tarball-fetcher.js
@@ -44 +44,5 @@ export default class TarballFetcher extends BaseFetcher {
+  constructor(dest: string, remote: PackageRemote, config: Config) {
+    super(dest, remote, config);
+  }
+
   validateError: ?Object = null;
--- a/node_modules/yarn/src/lockfile/parse.js
+++ b/node_modules/yarn/src/lockfile/parse.js
@@ -75 +75 @@ function* tokenise(input: string): Iterator<Token> {
-      if (nextNewline === -1) {
+      if (nextNewline < 0) {
@@ -388 +388,2 @@ function parse(str: string, fileLoc: string): Object {
+  let error: any;
   if (!fileLoc.endsWith(`.yml`)) {
@@ -391,10 +392,4 @@   if (!fileLoc.endsWith(`.yml`)) {
     } catch (error1) {
+      error = error1;
+  } }
       try {
-        return safeLoad(str, {
-          schema: FAILSAFE_SCHEMA,
-        });
-      } catch (error2) {
-        throw error1;
-      }
-    }
-  } else {
@@ -409 +404,3 @@   } else {
+      } catch (error2) {
+        throw error || error2;
   }
--- a/node_modules/yarn/src/package-hoister.js
+++ b/node_modules/yarn/src/package-hoister.js
@@ -938 +938 @@ export class NohoistResolver {
-  initNohoist = (info: HoistManifest, parent: ?HoistManifest) => {
+  initNohoist(info: HoistManifest, parent: ?HoistManifest) {
@@ -959,2 +959,2 @@ export class NohoistResolver {
     info.isNohoist = this._isNohoist(info);
-  };
+  }
@@ -970 +970 @@ export class NohoistResolver {
-  highestHoistingPoint = (info: HoistManifest): ?number => {
+  highestHoistingPoint(info: HoistManifest): ?number {
@@ -972,4 +972,4 @@ export class NohoistResolver {
-  };
+  }
 
   // private functions
-  _isNohoist = (info: HoistManifest): boolean => {
+  _isNohoist(info: HoistManifest): boolean {
@@ -986,2 +986,2 @@ export class NohoistResolver {
-  };
-  _isRootPackage = (pkg: Manifest): boolean => {
+  }
+  _isRootPackage(pkg: Manifest): boolean {
@@ -989,4 +989,4 @@ export class NohoistResolver {
-  };
-  _originalPath = (info: HoistManifest): string => {
+  }
+  _originalPath(info: HoistManifest): string {
     return this._makePath(info.originalParentPath, info.pkg.name);
-  };
+  }
@@ -998 +998 @@ export class NohoistResolver {
-  _isTopPackage = (info: HoistManifest): boolean => {
+  _isTopPackage(info: HoistManifest): boolean {
@@ -1005,4 +1005,4 @@ export class NohoistResolver {
-  };
-  _isLink = (info: HoistManifest): boolean => {
+  }
+  _isLink(info: HoistManifest): boolean {
     return info.pkg._remote != null && LINK_TYPES.has(info.pkg._remote.type);
-  };
+  }
@@ -1012 +1012 @@ export class NohoistResolver {
-  _extractNohoistList = (pkg: Manifest, pathPrefix: string): ?Array<string> => {
+  _extractNohoistList(pkg: Manifest, pathPrefix: string): ?Array<string> {
@@ -1019,2 +1019,2 @@ export class NohoistResolver {
     return nohoistList;
-  };
+  }
--- a/node_modules/yarn/src/resolvers/registries/npm-resolver.js
+++ b/node_modules/yarn/src/resolvers/registries/npm-resolver.js
@@ -27 +27,2 @@ type RegistryResponse = {
+  time?: ?{[key: string]: ?string},
 };
@@ -51 +52,16 @@ export default class NpmResolver extends RegistryResolver {
+    // Only install package version that have release date before a specified one.
+    // That date string must match JSON format, e.g. 2020-08-14T04:47:38.210Z
+    if (config.packageDateLimit && body.time) {
+      const releaseDates = body.time;
+      let closestVersion = null;
+      (semver: Object).rsort(Object.keys(body.versions)).some(v => {
+        if (releaseDates[v] && semver.satisfies(v, range)) {
+          closestVersion = v;
+          if (releaseDates[v] < config.packageDateLimit) return true;
+        }
+        return false;
+      });
+      if (closestVersion) return body.versions[closestVersion];
+    }
+
     // If the latest tag in the registry satisfies the requested range, then use that.
@@ -97 +113 @@ export default class NpmResolver extends RegistryResolver {
-    const body = await this.config.registries.npm.request(escapedName);
+    const body = await this.config.registries.npm.request(escapedName, {unfiltered: !!this.config.packageDateLimit});
--- a/node_modules/yarn/src/util/dynamic-require.js
+++ b/node_modules/yarn/src/util/dynamic-require.js
@@ -4 +4 @@
-exports.dynamicRequire = typeof __webpack_require__ !== 'undefined' ? __non_webpack_require__ : require; // eslint-disable-line
+export const dynamicRequire = typeof __webpack_require__ != 'undefined' ? __non_webpack_require__ : require; // eslint-disable-line
--- a/node_modules/yarn/src/util/execute-lifecycle-script.js
+++ b/node_modules/yarn/src/util/execute-lifecycle-script.js
@@ -237 +237,2 @@ export async function makeEnv(
+  if (!config.disableWrappersFolder)
   pathParts.unshift(await getWrappersFolder(config));
--- a/node_modules/yarn/src/util/promise.js
+++ b/node_modules/yarn/src/util/promise.js
@@ -12,6 +12,3 @@     return new Promise(function(resolve, reject) {
-      args.push(function(err, ...result) {
-        let res = result;
+      args.push(function(err) {
+        let res: any;
 
-        if (result.length <= 1) {
-          res = result[0];
-        }
@@ -21 +18,5 @@         if (firstData) {
           err = null;
+        } else if (arguments.length <= 2) {
+          res = arguments[1];
+        } else {
+          res = Array.prototype.slice.call(arguments, 1);
--- a/node_modules/yarn/src/util/semver.js
+++ b/node_modules/yarn/src/util/semver.js
@@ -3 +3,2 @@
-import semver, {type Release} from 'semver';
+import type {Release} from 'semver';
+const semver = require('semver');
