--- a/node_modules/ansi-styles/index.js
+++ b/node_modules/ansi-styles/index.js
@@ -162,4 +162 @@ function assembleStyles() {
-Object.defineProperty(module, 'exports', {
-	enumerable: true,
-	get: assembleStyles
-});
+module.exports = assembleStyles();
--- a/node_modules/glob/glob.js
+++ b/node_modules/glob/glob.js
@@ -46 +46 @@ var Minimatch = minimatch.Minimatch
-var inherits = require('inherits')
+var inherits = require('util').inherits
--- a/node_modules/string_decoder/lib/string_decoder.js
+++ b/node_modules/string_decoder/lib/string_decoder.js
@@ -26 +26 @@
-var Buffer = require('safe-buffer').Buffer;
+var Buffer = require('buffer').Buffer;
@@ -104 +104 @@ function StringDecoder(encoding) {
-  this.lastChar = Buffer.allocUnsafe(nb);
+  this.lastChar = (Buffer.allocUnsafe || Buffer)(nb);
