--- a/node_modules/deep-equal/index.js
+++ b/node_modules/deep-equal/index.js
@@ -1,5 +1,5 @@
-var objectKeys = require('object-keys');
+var objectKeys = Object.keys;
 var isArguments = require('is-arguments');
-var is = require('object-is');
+var is = typeof Object.is == 'function' ? Object.is : require('object-is');
 var isRegex = require('is-regex');
-var flags = require('regexp.prototype.flags');
+var flags = Function.call.bind((/./mi.flags === 'im' && Object.getOwnPropertyDescriptor(RegExp.prototype, 'flags').get) || require('regexp.prototype.flags/implementation'));
--- a/node_modules/glob/glob.js
+++ b/node_modules/glob/glob.js
@@ -45 +45,0 @@ var minimatch = require('minimatch')
-var Minimatch = minimatch.Minimatch
@@ -56 +55,0 @@ var inflight = require('inflight')
-var util = require('util')
--- a/node_modules/glob/sync.js
+++ b/node_modules/glob/sync.js
@@ -6,3 +6,0 @@ var minimatch = require('minimatch')
-var Minimatch = minimatch.Minimatch
-var Glob = require('./glob.js').Glob
-var util = require('util')
--- a/node_modules/is-core-module/index.js
+++ b/node_modules/is-core-module/index.js
@@ -3 +3 @@
-var has = require('has');
+var has = Function.call.bind(Object.prototype.hasOwnProperty);
--- a/node_modules/is-regex/index.js
+++ b/node_modules/is-regex/index.js
@@ -3,2 +3,2 @@
-var callBound = require('call-bind/callBound');
-var hasToStringTag = require('has-tostringtag/shams')();
+var callBound = function (fn) { return Function.call.bind(fn); };
+var hasToStringTag = typeof Symbol == 'function' && !!Symbol.toStringTag;
@@ -11,2 +11,2 @@ if (hasToStringTag) {
-	has = callBound('Object.prototype.hasOwnProperty');
-	$exec = callBound('RegExp.prototype.exec');
+	has = callBound(Object.prototype.hasOwnProperty);
+	$exec = callBound(RegExp.prototype.exec);
@@ -28 +28 @@ if (hasToStringTag) {
-var $toString = callBound('Object.prototype.toString');
+var $toString = callBound(Object.prototype.toString);
--- a/node_modules/regexp.prototype.flags/implementation.js
+++ b/node_modules/regexp.prototype.flags/implementation.js
@@ -3 +3,0 @@
-var functionsHaveConfigurableNames = require('functions-have-names').functionsHaveConfigurableNames();
@@ -37 +36 @@ module.exports = function flags() {
-if (functionsHaveConfigurableNames && Object.defineProperty) {
+if (module.exports.name && Object.getOwnPropertyDescriptor(module.exports, 'name').configurable) {
--- a/node_modules/tape/lib/results.js
+++ b/node_modules/tape/lib/results.js
@@ -9,3 +9,3 @@ var inspect = require('object-inspect');
-var callBound = require('call-bind/callBound');
-var has = require('has');
-var $exec = callBound('RegExp.prototype.exec');
+var callBound = function (fn) { return Function.call.bind(fn); };
+var has = callBound(Object.prototype.hasOwnProperty);
+var $exec = callBound(RegExp.prototype.exec);
--- a/node_modules/tape/lib/test.js
+++ b/node_modules/tape/lib/test.js
@@ -8,10 +8,10 @@ var EventEmitter = require('events').EventEmitter;
-var has = require('has');
+var callBound = function (fn) { return Function.call.bind(fn); };
+var has = callBound(Object.prototype.hasOwnProperty);
 var isRegExp = require('is-regex');
-var trim = require('string.prototype.trim');
-var callBound = require('call-bind/callBound');
-var forEach = require('for-each');
+var trim = callBound(String.prototype.trim);
+var forEach = callBound(Array.prototype.forEach);
 var inspect = require('object-inspect');
-var isEnumerable = callBound('Object.prototype.propertyIsEnumerable');
-var toLowerCase = callBound('String.prototype.toLowerCase');
-var $exec = callBound('RegExp.prototype.exec');
-var objectToString = callBound('Object.prototype.toString');
+var isEnumerable = callBound(Object.prototype.propertyIsEnumerable);
+var toLowerCase = callBound(String.prototype.toLowerCase);
+var $exec = callBound(RegExp.prototype.exec);
+var objectToString = callBound(Object.prototype.toString);
