--- a/node_modules/terser-es/lib/output.js
+++ b/node_modules/terser-es/lib/output.js
@@ -1743,2 +1743,2 @@ function OutputStream(options) {
-                output.comma();
-                if (output.should_break()) {
+                output.print(",");
+                if (!output.should_break()) output.space(); else {
--- a/node_modules/terser-es/lib/sourcemap.js
+++ b/node_modules/terser-es/lib/sourcemap.js
@@ -46 +46 @@
-import MOZ_SourceMap from "source-map";
+import MOZ_SourceMap from "../source-map";
--- a/node_modules/terser-es/lib/compress/index.js
+++ b/node_modules/terser-es/lib/compress/index.js
@@ -249 +249,2 @@ class Compressor extends TreeWalker {
             keep_infinity : false,
+            lhs_constants : !false_by_default,
@@ -5823,2 +5824,2 @@ def_optimize(AST_Binary, function(self, compressor) {
-    if (commutativeOperators.has(self.operator)) {
+    if (compressor.option("lhs_constants") && commutativeOperators.has(self.operator)) {
         if (self.right.is_constant()
@@ -5852,2 +5853,4 @@ def_optimize(AST_Binary, function(self, compressor) {
             self.left = make_node(AST_Null, self.left);
+        } else if (!is_strict_comparison && is_undefined(self.right, compressor)) {
+            self.right = make_node(AST_Null, self.right);
         } else if (compressor.option("typeofs")
@@ -5866 +5869,13 @@ def_optimize(AST_Binary, function(self, compressor) {
+        } else if (compressor.option("typeofs")
+            && self.left instanceof AST_UnaryPrefix
+            && self.left.operator == "typeof"
+            && self.right instanceof AST_String
+            && self.right.value == "undefined") {
+            var expr = self.left.expression;
+            if (expr instanceof AST_SymbolRef ? expr.is_declared(compressor)
+                : !(expr instanceof AST_PropAccess && compressor.option("ie8"))) {
+                self.left = expr;
+                self.right = make_node(AST_Undefined, self.right).optimize(compressor);
+                if (self.operator.length == 2) self.operator += "=";
+            }
         } else if (self.left instanceof AST_SymbolRef
--- a/node_modules/terser-es/tools/terser.d.ts
+++ b/node_modules/terser-es/tools/terser.d.ts
@@ -45 +45,2 @@ export interface CompressOptions {
     keep_infinity?: boolean;
+    lhs_constants?: boolean;
